### PR TITLE
Add support for releasing to Gradle Plugin Portal

### PR DESCRIPTION
**Pull request by cmelchior**
This PR adds support for releasing the `io.realm.kotlin` plugin to Gradle Plugin Portal. Uploading works locally and I'm just waiting for some manual verification on their end. Before the plugin was called `realm-kotlin`. This is arguably a better-looking and shorter name, but Gradle Plugin Portal requires that plugins follow the normal hierarchy structure as on Maven Central, so I renamed it to `io.realm.kotlin`. If anyone has a better idea, let me know.

TODO:
- [x] Ensure that the manually uploaded plugin works.

### CODE REVIEW

**Comment by cmelchior** [comment]
Current plugin got rejected:

```
The new submission has outstanding issue:
Could you get rid of the leading "gradle.plugin" in the artifact coordinates. You can do this by following the full example at https://plugins.gradle.org/docs/publish-plugin
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,18 @@
+## 0.2.0 (YYYY-MM-DD)
+
+### Breaking Changes
+* The Realm Kotlin Gradle plugin has changed name from `realm-kotlin` to `io.realm.kotlin`.
```
Comment (nhachicha) [inline CHANGELOG.md:4]:
```suggestion
* The Realm Kotlin Gradle plugin has changed name from `realm-kotlin` to `io.realm.kotlin` to align with Gradle Plugin Portal requirements.
```

File: examples/kmm-sample/androidApp/build.gradle.kts [hunk 2]
```diff
@@ -20,36 +20,16 @@ plugins {
     id("com.android.application")
     kotlin("android")
     id("kotlin-android-extensions")
-    // TODO Publish marker artifact to OJO to allow applying plugin by id
-    //  https://github.com/realm/realm-kotlin/issues/100
     // Apply Realm Kotlin plugin even though we technically do not need it, to ensure that we have
     // the right kotlinOptions
-    // id("realm-kotlin") version Realm.version
+    id("io.realm.kotlin") version Realm.version
```
Comment (nhachicha) [inline examples/kmm-sample/androidApp/build.gradle.kts:25]: 😻

File: examples/kmm-sample/build.gradle.kts [hunk 3]
```diff
@@ -13,20 +12,10 @@ buildscript {
...
-//    }
-//    repositories {
-//        maven(url = "http://oss.jfrog.org/artifactory/oss-snapshot-local")
-//    }
-// }
-repositories {
-    mavenCentral()
+allprojects {
+    repositories {
+        gradlePluginPortal()
```
Comment (nhachicha) [inline examples/kmm-sample/build.gradle.kts:17]: Do we need to explicitly add `gradlePluginPortal` ?

File: packages/gradle-plugin/build.gradle.kts [hunk 4]
```diff
@@ -32,19 +34,27 @@ dependencies {
...
 
+fun createMarkerArtifact(): Boolean {
+    val value = properties.getOrDefault("generatePluginArtifactMarker", "false") as String
+    return value.toBoolean()
+}
+
+pluginBundle {
+    website = "https://github.com/realm/realm-kotlin"
+    vcsUrl = "https://github.com/realm/realm-kotlin"
+    tags = listOf("MongoDB", "Realm", "Database", "Kotlin", "Mobile", "Multiplatform", "Android")
```
Comment (nhachicha) [inline packages/gradle-plugin/build.gradle.kts:45]: I would add "KMM" as a key word

**Review by nhachicha — APPROVED** [review]
Nice and clean 🚀

File: examples/kmm-sample/build.gradle.kts (hunk 3 above)
Comment (cmelchior) [reply examples/kmm-sample/build.gradle.kts:17]: No, removed

**Pull request merged**